python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1385,31 +1394,40 @@ def test_torch_tensor_nccl_comm_deduplicate_collective_and_p2p(ray_start_regular
         ]
         collectives = collective.allreduce.bind(computes)
         recvs = [
-            workers[(i + 1) % num_workers].recv.bind(
-                collective.with_type_hint(TorchTensorType(transport="nccl"))
-            )
-            for i, collective in enumerate(collectives)
+            # Each of the 2 worker recvs from the other.
+            workers[0].recv.bind(
+                collectives[1].with_type_hint(TorchTensorType(transport="nccl"))
+            ),
+            workers[1].recv.bind(
+                collectives[0].with_type_hint(TorchTensorType(transport="nccl"))
+            ),
         ]
         dag = MultiOutputNode(recvs)
 
     compiled_dag = dag.experimental_compile()
 
     from ray.dag.collective_node import CollectiveOutputNode
+    from ray.experimental.channel import ChannelContext
 
-    comms = set()
+    comm_ids = set()
     for _, exec_tasks in compiled_dag.actor_to_executable_tasks.items():
         for exec_task in exec_tasks:
             dag_node = compiled_dag.idx_to_task[exec_task.task_idx].dag_node
             if isinstance(dag_node, CollectiveOutputNode):
                 assert exec_task.collective_group is not None
                 nccl_group_id = exec_task.collective_group._type_hint._nccl_group_id
```

Comment:
_type_hint should be public as type_hint. Also for _nccl_group_id.

Reply:
ack